R/shiny.R

```diff
@@ -1,31 +1,38 @@
 #' Internal function that expose javascript bindings to Shiny app.
 .onLoad <- function(libname, pkgname) {
   # Add directory for static resources
-  message("attaching resources")
   shiny::addResourcePath('shiny.semantic', system.file('www', package = 'shiny.semantic', mustWork = TRUE))
 }
 
-#' Semantic UI page
-#'
-#' Universal binding for Shiny input on custom user interface. Using this function one can create various inputs ranging from text, numerical, date, dropdowns, etc. Value of this input is extracted via jQuery using $().val() function and exposed as serialized JSON to the Shiny server.
+#' Universal binding for Shiny input on custom user interface. Using this function one can create various inputs
+#' ranging from text, numerical, date, dropdowns, etc. Value of this input is extracted via jQuery using $().val()
+#' function and default exposed as serialized JSON to the Shiny server. If you want to change type of exposed input
+#' value specify it via type param. Currently list of supported types is "JSON" (default) and "text".
 #'
 #' @param input_id String with name of this input. Access to this input within server code is normal with input[[input_id]].
 #' @param shiny_ui UI of HTML component presenting this input to the users. This UI should allow to extract its value with jQuery $().val() function.
 #' @param value An optional argument with value that should be set for this input. Can be used to store persisten input valus in dynamic UIs.
+#' @param type Type of input value (could be "JSON" or "text").
 #'
 #' @export
-shiny_input <- function(input_id, shiny_ui, value = NULL) {
+shiny_input <- function(input_id, shiny_ui, value = NULL, type = "JSON") {
```

Comment:
i think it would make sense to introduce `shiny_text_input` convenience function that would call the general `shiny_input` with `text`, WDYT?